### PR TITLE
feat: enable remix feature flags

### PR DESCRIPTION
**Pull request by JohnnyAir**
Enabled remix future feature flags.
https://remix.run/docs/en/main/start/future-flags

### CODE REVIEW

File: app/routes/organizations_.$organizationSlug._index.tsx
Comment (janhesters) [inline app/routes/organizations_.$organizationSlug._index.tsx:1]: Why are we deleting the layout pages?

**Review by janhesters — APPROVED** [review]

File: app/routes/organizations_.$organizationSlug._index.tsx
Comment (JohnnyAir) [reply app/routes/organizations_.$organizationSlug._index.tsx:1]: Redirecting on index routes leads to a waterfall effect.

In single fetch mode, Remix initially calls the layout and index loaders and waits for a response. If a redirect to another sub-route is encountered, Remix then loads the layout along with the sub-route loader.

By removing the index page—if it is intended to always redirect to the default page—we can help avoid the waterfall effect. This approach allows us to move the check to the layout loader, preventing data loading when it isn't the base route.

File: app/routes/organizations_.$organizationSlug.tsx [hunk 1]
```diff
@@ -1,11 +1,30 @@
...
 
 export const handle = { i18n: ['organizations', 'sidebar', 'header'] };
 
+/* 
+  With single fetch enabled, the layout route loader will always be called for every subroute change.
+  we can skip re-running this loader if the organization slug remains unchanged by opting into Remix's granular single fetch.
+  If you want to opt out of granular single fetch and always re-run this loader, you can remove this function
+  To learn more about granular single fetch, see the Remix documentation on revalidations: 
+  https://remix.run/docs/en/main/guides/single-fetch#revalidations
+ **/
```
Comment (janhesters) [inline app/routes/organizations_.$organizationSlug.tsx:16]: This sucks!

File: tsconfig.json [hunk 2]
```diff
@@ -17,7 +17,8 @@
     "resolveJsonModule": true,
     "skipLibCheck": true,
     "strict": true,
-    "target": "ES2022"
+    "target": "ES2022",
+    "types": ["@remix-run/node", "vite/client"]
```
Comment (janhesters) [inline tsconfig.json:21]: This sucks!

File: tsconfig.json [hunk 3]
```diff
@@ -17,7 +17,8 @@
     "resolveJsonModule": true,
     "skipLibCheck": true,
     "strict": true,
-    "target": "ES2022"
+    "target": "ES2022",
```
Comment (janhesters) [inline tsconfig.json:20]: This sucks!

**Pull request merged**